src/openMVG/features/akaze/AKAZE.cpp

```diff
@@ -1,8 +1,5 @@
-// Copyright (c) 2014 Romuald PERROT, Pierre MOULON.
```

Comment:
Please note that you can't remove a copyright.
Removing a copyright on any file is not fair towards the OpenSource community.

According the [MPL2 license](https://www.mozilla.org/en-US/MPL/2.0/FAQ/):
>3.4. Notices
You may not remove or alter the substance of any license notices (including copyright notices, patent notices, disclaimers of warranty, or limitations of liability) contained within the Source Code Form of the Covered Software, except that You may alter any license notices to the extent required to remedy known factual inaccuracies.

Reply:
The ambition was not to alter it but to clean the source code. We would like to avoid management of copyright per file which is difficult to maintain and was already quite incomplete. We would like to manage such things at the project level.
But in all case, don't be afraid, "the use of copyright notices has become optional to claim copyright, because the Berne Convention makes copyright automatic".

Reply:
Clean does not mean remove the copyright information, else it is too easy to claim ownership.

It does not change the fact that we have to respect the MPL2 License.
According the MPL2 3.4 notice you are enforced to keep the copyright information even at **source code** level.

Reply:
> "You may not remove or alter **the substance** of any license notices".

We don't think that we alter the substance by maintaining it globally. We are trying to make it simpler, easier to understand, easier to maintain and also "remedy known factual inaccuracies".

We think that it's very difficult to maintain it per file and we don't see any good reason to do so. We are convinced that it leads to many errors and also to endless discussions.
As an example, we consider that most of the copyright currently in openMVG are incorrect. All the work you have done during the period of your CIFRE contract should be copyrighted to Mikros Image and Imagine ENPC. "If the work was created by an employee in the course of his or her duties, the employer is the sole owner of the copyright".
But we would prefer to move to global licensing and avoid such useless discussions. So You, Mikros Image, Imagine ENPC are recognized as contributors of the project, we don't need to go file by file or function by function.

This process of clarification and simplification has been motivated by the setup of a global pipeline with a new name, without denying the principles of collaborative work and legacy which are of great values for all of us.

Reply:
@fabiencastan Please be aware that the license with the copyright mention have been chosen by the three involved parties (Mikros Image, Imagine and Me)!

See the initial release and first commit:
https://github.com/openMVG/openMVG/commit/5c5c221ca8e93a58eb1d1dd83394f75b5f43f48b#diff-896a14d7505334a5cd5d1298fa269da5R1
And this https://github.com/openMVG/openMVG/commit/5c5c221ca8e93a58eb1d1dd83394f75b5f43f48b#diff-04c6e90faac2675aa89e2176d2eec7d8R45

I agree that a large part of OpenMVG have been created during my CIFRE contract but the three party agreed on the license.

Please also note that a lot of modifications have been done since this time under no company leader ship so it remains affiliated only to the OpenMVG community and authors copyright as the license authorized it.

The main reason of existence of the MPL2 is to be file only in order to be to protect HEADER file library. Other type of licenses does not have this clause!

- Moving to a global licensing can be done, with the agreement of the other persons Licensed code.

Briefly, make sure that you do not break some third party licensed code.

Reply:
MPL2 is explicitly a per-file license, from "1.4 'Covered software' means Source Code Form to which the initial Contributor has attached the notice in Exhibit A, [...]". In my interpretation of this paragraph, the presence of the copyright notice in a source file is the only thing that makes it MPLv2 and this cannot be replaced by a "global" copyright notice.

The substance is not changed by being maintained globally, but paragraph 3.4 still forbids removal of the notice from the source file.

Reply:
@pmoulon 

> Please be aware that the license with the copyright mention have been chosen by the three involved parties (Mikros Image, Imagine and Me)!

I was not there to know what has been discussed, but the copyright is inalienable unless you have a document from Mikros Image and Imagine ENPC saying that they transfer their rights.

> Briefly, make sure that you do not break some third party licensed code.

Thanks for your understanding. I hope we will manage to continue exchanges and discussions to foster improvements between the 2 forks.

@zvrba 

> MPL2 is explicitly a per-file license

Yes, sure. It's often called file-based license as it protects the Source Code in opposition to the GPL family which focus on the protection of the binaries generated from this Source Code (so for instance building in static or dynamic is not the same in the GPL family while this notion doesn't exist with the MPL).

> "1.4 'Covered software' means Source Code Form to which the initial Contributor has attached the notice in Exhibit A, [...]"

Yes, but if you read the definition of "Exhibit A" it says that "If it is not possible or desirable to put the notice in a particular file, then You may include the notice in a location (such as a LICENSE file in a relevant directory) where a recipient would be likely to look for such a notice."

> the presence of the copyright notice in a source file is the only thing that makes it MPLv2 and this cannot be replaced by a "global" copyright notice

Copyright and license are 2 different things.
From the FAQ Q4: "You may also add additional accurate notices of copyright ownership, such as the name of the copyright holder, but this is not necessary."

> the presence of the copyright notice in a source file is the only thing that makes it MPLv2 and this cannot be replaced by a "global" copyright notice

From the FAQ: "Q25: What happens if someone doesn't use the per-file boilerplate, and just ships a copy of the full MPL 2 with their code? The code is licensed under the plain MPL 2."

An a more explicit and detailed answer in FAQ: "Q22: Does MPL 2.0 require that the MPL 2.0 license notice header be included in every file?
The license notice must be in some way "attached" to each file. (Sec. 1.4.) In cases where putting it in the file is impossible or impractical, that requirement can be fulfilled by putting the notice somewhere that a recipient "would be likely to look for such a notice," such as a LICENSE file in the same directory as the file. (Ex. A.) The license notice is as short as possible (3 lines) to make it easy to put in as many different types of files as possible. While the license permits putting the header somewhere other than the file itself, individual files often end up being distributed on their own, without the rest of the software they were authored with. As a result, putting the license notice in the file is the surest way to ensure that recipients are always notified."

And we still have an explicit mention of the MPL in each file. The intention in changing this sentence is to explicitly say which file is part of the project (as we include some third-dependencies directly in the source tree).

So I don't think there is an infringement, but you are right that it is explicitly recommended to use the text from "Exhibit A", so it is maybe a better idea to follow this recommendation.

Reply:
@fabiencastan 
> I was not there to know what has been discussed, but the copyright is inalienable unless you have a document from Mikros Image and Imagine ENPC saying that they transfer their rights.

1. MikrosImage released the project under the specified license (I don't need to prove it) http://opensource.mikrosimage.eu/openmvg.html 

2. As you said a `copyright is inalienable`.  So, please give some respect to the owner of the modification and do not alter or `remove` their copyright.

Please note that when you cite Q4, you forgot to mention that the statement is true only if you have written the code. In this case you try to apply some changes on code you did not write. `Q4: I want to use the Mozilla Public License for software that I have written. What do I have to do?`

Finally, if you want to say that you want claim a copy under the copyright of MikrosImage or Imagine ENPC you can do that only for the code I wrote as an employee, so it would limit your fork to OpenMVG v0.3 https://github.com/openMVG/openMVG/releases/tag/v0.3. Anything that have been made later is under the copyright of the owner of the files!

Reply:
But. MPL is a viral license, i.e., source code that has at some time been released as MPL, stays under MPL. For copyright purposes, does the authorship / the original contributor really matter as long as the file in question can be traced to its MPL origin?

"Exhibit A" from the license text does not contain placeholder for the author(s).

Reply:
Tracing does not allows you to remove a license or a copyright.

I mean: Will you be ok to remove a license form a project because you can see it in the git history!

In some case the authorship cannot be traced since I have collaborated with some persons (Romuald Perrot, Pascal Monasse, Lionel Moisan, Stephane Flotron and others...), so we used the copyright information to show that we work as a group on some functionnalties.

In the case you don't care about authorship, original contributor, it's not being fair towards the opensource community. 

I would like to say it's like to remove the author page of a book and say the author page can be found somewhere in an archive. -> So: 1: it makes the person confuse about the origin of the file, 2: it's not respectful to the author(s).

> "Exhibit A" from the license text does not contain placeholder for the author(s).

As I said previously, the copyright is an option, but it does not mean that if it exits you can remove it.
You can see that other project (Eigen) are using the copyright mention too: https://bitbucket.org/eigen/eigen/src/e99c823108f4e1cfabd27b083501f465ec98b40d/Eigen/Core?at=default&fileviewer=file-view-default#Core-4

Reply:
Again, we are not removing the name of anybody from the project (or the book).

The reality is that the copyrights per file are largely inaccurate which also creates confusion. On one hand, it seems impossible to fix it in all files and on the other hand you refuse to see it managed globally. So I don't know what to do with that.

Reply:
@fabiencastan Compromise solution? Use a generic MPL2 header (w/o mentioning contributors) in source files, and have a list of contributors in a global copyright file.

> The reality is that the copyrights per file are largely inaccurate 

MPL explicitly allows changing of copyright notices to correct factual errors.  Since MPL2 doesn't seem to care about who the original contributor is (re "Exhibit A"), I think that maintaining a global list of contributors doesn't breach any copyright.

Reply:
I agree. I updated all the files to the standard MPL2 header [here](https://github.com/alicevision/AliceVision/pull/336/commits/775601bd04a89289fa2d56017eb11c702959c6c8).

Reply:
> The reality is that the copyrights per file are largely inaccurate

I would to know why and how you can argue on this!
For me removing the copyright is not fixing any factual errors.

Reply:
As discussed before, Mikros Image and Imagine should be part of the copyright of OpenMVG v0.3. Then the work done in our fork in the last 3 years by IRIT, Mikros Image and Simula should be part of it, but it is very difficult to decide per file when a partner has done significant enough changes to become part of it.

Reply:
Partners agreed with the existing license, copyright and source code release.

Then if people are making commits, they are agreeing the existing license and copyright. This is how open source works:
- you agreed with the existing license and copyright.

Reply:
Hi. Gerv here; I'm with Mozilla, and answer questions about the MPL2.

In general, you can't go removing people's copyright lines from a file, even if you think it would be easier to manage that way. Section 3.4 says you can't. The other text you quote is about whether you have to or want to add them in the first place; but once a copyright holder has put them there, they can't be changed or removed without that person's permission.

If the copyright notices are inaccurate, that's a different question; in that case, they could be altered (to make them accurate) or removed. It's not for me to judge any claims of inaccurate copyright notices. But this is the only permissible reason to remove or alter someone else's copyright notice.

It is fine for a copyright notice to name some, but not all, copyright holders. Copyright is automatic, and so not being named doesn't affect your rights. But that automatic-ness doesn't mean you can remove notices (section 3.4 again).

Reply:
Thanks @gerv for the clarification.
Yes, we consider that the initial copyright was inaccurate as it should mention Mikros Image and Image ENPC.

Reply:
If it's inaccurate for that reason, you are allowed to fix it by adding those two companies. You are not allowed to remove existing copyright information unless that copyright information is incorrect.

"It's inaccurate in some way" does not give you a blank cheque to make any changes to the copyright information you like.

Reply:
@fabiencastan Please be aware that your assumption is wrong. 

1. The listed person in the copyright was the creator of those files and I remember you again that Mikros Image and Imagine ENPC agreed on this license and copyright information!
2. Many of the files, (most of them), have been done out of any company context and so you must keep the existing copyright and you don't have any right to remove it! (As I told you, anything after OpenMVG v0.3 is out of MikrosImage and Imagine ENPC context).

Note that, for example, this particular file I pointed you `openMVG/features/akaze/AKAZE.cpp` have been created in 2014. So this code does not have any relation with MikrosImage or Imagine ENPC!

A similar case exists for many other files (i.e. openMVG/cameras, sfm, graph, matching_image_collection, ...).

So as @gerv explained to you, please follow the rules and be fair to the community.

It's too easy to claim that it's not accurate without knowing the context (date, or company agreement), else you can claim that for any open source project.... and the open source world would not be a safe place to gather and work in a collaborative way.

Reply:
It is true that erroneously removing correct copyright information would be a license violation, and would lead to loss of all rights under the license (section 5.1). Therefore, if there is any doubt, it is best to leave it in place.